src/Model/Item.php

```diff
@@ -290,4 +291,135 @@ public static function addShadowPost($itemid)
 
 		logger("Stored public shadow for comment ".$item['uri']." under id ".$public_shadow, LOGGER_DEBUG);
 	}
+
+	 /**
+	 * Adds a "lang" specification in a "postopts" element of given $arr,
+	 * if possible and not already present.
+	 * Expects "body" element to exist in $arr.
+	 *
+	 * @todo change to "local", once the "Item::insert" is in this class
+	 */
+	public static function addLanguageInPostopts(&$arr)
+	{
+		if (x($arr, 'postopts')) {
+			if (strstr($arr['postopts'], 'lang=')) {
+				// do not override
+				return;
+			}
+			$postopts = $arr['postopts'];
+		} else {
+			$postopts = "";
+		}
+
+		$naked_body = preg_replace('/\[(.+?)\]/','', $arr['body']);
+		$l = new Text_LanguageDetect();
+		$lng = $l->detect($naked_body, 3);
+
+		if (sizeof($lng) > 0) {
+			if ($postopts != "") {
+				$postopts .= '&'; // arbitrary separator, to be reviewed
+			}
+
+			$postopts .= 'lang=';
+			$sep = "";
+
+			foreach ($lng as $language => $score) {
+				$postopts .= $sep . $language . ";" . $score;
+				$sep = ':';
+			}
+			$arr['postopts'] = $postopts;
+		}
+	}
+
+	/**
+	 * @brief Creates an unique guid out of a given uri
+	 *
+	 * @param string $uri uri of an item entry
+	 * @param string $host (Optional) hostname for the GUID prefix
+	 * @return string unique guid
+	 */
+	public static function GuidFromUri($uri, $host = "")
```

Comment:
Standards: Please fix the capitalization of this method's name (no capital first letter)

Reply:
Sorry, it's just like we do on work ... (Is now fixed)